src/ControlSystem/TimescaleTuner.hpp

```diff
@@ -0,0 +1,42 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cmath>
+
+#include "DataStructures/DataVector.hpp"
+#include "ErrorHandling/Error.hpp"
+
+/// \ingroup ControlSystemGroup
+/// Manages control system timescales
+class TimescaleTuner {
+ public:
+  TimescaleTuner(DataVector initial_timescale, double max_timescale,
+                 double min_timescale, double max_threshold,
+                 double min_threshold, double increase_factor,
+                 double decrease_factor) noexcept;
+
+  TimescaleTuner(TimescaleTuner&&) noexcept = default;
+  TimescaleTuner& operator=(TimescaleTuner&&) noexcept = default;
+  TimescaleTuner(const TimescaleTuner&) = delete;
+  TimescaleTuner& operator=(const TimescaleTuner&) = delete;
+  ~TimescaleTuner() = default;
+
+  /// returns the current timescale for each component of a FunctionOfTime
+  DataVector current_timescale() const noexcept { return timescale_; }
+  /// manually sets all timescales to a specified value, if allowable
+  void suggested_timescale_if_in_allowable_range(double timescale) noexcept;
+  /// general function for internally updating timescales
```

Comment:
This documentation does not give me any idea about what this function does, except to suggest that it is for internal use (which I assume it is not).